File: python/src/sec_edgar_toolkit/core/form_10k.py
"""Structured view of a 10-K annual report, returned by ``Filing.obj()``."""

from __future__ import annotations

from ..parsers.items import TenKItem
from .periodic_report import PeriodicReport


class TenK(PeriodicReport):
    """10-K annual report: named sections and XBRL-backed financials."""

    SECTION_ITEMS = {
        "business": TenKItem.BUSINESS,
        "risk_factors": TenKItem.RISK_FACTORS,
        "mda": TenKItem.MANAGEMENT_DISCUSSION_AND_ANALYSIS,
    }

    def __repr__(self) -> str:
        return f"TenK(sections={list(self._items.keys())})"

File: typescript/src/core/form-10k.ts
/**
 * Structured view of a 10-K annual report, returned by `Filing.obj()`.
 */

import { TenKItem } from "../parsers/items";
import { PeriodicReport, ReportFiling } from "./periodic-report";

/** 10-K annual report: named sections and XBRL-backed financials. */
export class TenK extends PeriodicReport {
  static SECTION_ITEMS: Record<string, string> = {
    business: TenKItem.BUSINESS,
    riskFactors: TenKItem.RISK_FACTORS,
    mda: TenKItem.MANAGEMENT_DISCUSSION_AND_ANALYSIS,
  };

  static async create(filing: ReportFiling): Promise<TenK> {
    let items: Record<string, string> = {};
    try {
      items = await filing.extractItems();
    } catch {
      items = {};
    }
    return new TenK(filing, items);
  }
}
